src/cascadia/TerminalSettingsModel/ProfileEntry.cpp

```diff
@@ -0,0 +1,41 @@
+#include "pch.h"
+#include "ProfileEntry.h"
+#include "JsonUtils.h"
+
+#include "ProfileEntry.g.cpp"
+
+using namespace Microsoft::Terminal::Settings::Model;
+using namespace winrt::Microsoft::Terminal::Settings::Model::implementation;
+
+static constexpr std::string_view ProfileKey{ "profile" };
+
+ProfileEntry::ProfileEntry() noexcept :
+    ProfileEntry{ winrt::hstring{} }
+{
+}
+
+ProfileEntry::ProfileEntry(const winrt::hstring& profile) noexcept :
+    ProfileEntryT<ProfileEntry, NewTabMenuEntry>(NewTabMenuEntryType::Profile),
+    _ProfileName{ profile }
+{
+}
+
+Json::Value ProfileEntry::ToJson() const
+{
+    auto json = NewTabMenuEntry::ToJson();
+
+    // We always store the GUID of the profile since that is less
+    // errorprone in the long term.
+    JsonUtils::SetValueForKey(json, ProfileKey, _Profile.Guid());

```

Comment:
Hmm... What happens if the user renames a profile? We/You need to be sure to test this with renaming via the Settings UI.

ToJson only gets called when [Save] button is pressed via the Settings UI, so I guess when somebody modifies the JSON, it doesn't really matter (this code isn't executed).

Reply:
So when the settings load, the matching between textual value and profile instance is done. When the user edits the JSON and changes the name but does not change the menu entry, it will be a broken link that will be ignored. When the user changes a name in the UI, the link in memory between menu entry and profile is still in place, so when saving to JSON the GUID will be used to guarantee future linking.